python/lsst/ap/association/diaPipe.py

```diff
@@ -239,9 +241,14 @@ def run(self, diaSourceCat, diffIm, exposure, ccdExposureIdBits):
             exposure.getInfo().getVisitInfo().getDate().toPython())
         self.apdb.storeDiaForcedSources(diaForcedSources)
         if self.config.doPackageAlerts:
+            diaForcedSources.set_index("diaObjectId", drop=False, inplace=True)
+            if len(loaderResult.diaForcedSources) > 1:
+                diaForcedSources = pd.concat([diaForcedSources,
+                                              loaderResult.diaForcedSources])
```

Comment:
I'm kind of surprised by the order here, I'd have guessed that you need to `set_index` _after_ doing the `concat` with `loaderResult`. Maybe this is fine, I don't really know, just mentioning it.

Reply:
Oh yeah, I see the same index is set in `loadDiaForcedSources`, which I guess is where this value comes from, right?

Reply:
Yeah, what you're seeing here was an attempt to get around the fact that when joined, dataframes can do weird things with the types. Now that you mention it, I'm not sure the data ends up sorted here. I'll switch back to an appended with a specific "sort" to make this more explicit.

Reply:
I moved the diaForcedSources indexing into the diaForcedSource task to keep it more compartmentalized.